src/commands/registerCommands.ts

```diff
@@ -0,0 +1,87 @@
+import * as path from 'path';
+import { commands, ExtensionContext, Position, Uri, window, workspace } from "vscode";
+import { CancellationToken, ExecuteCommandParams, ExecuteCommandRequest, LanguageClient, ReferencesRequest, TextDocumentIdentifier } from "vscode-languageclient";
+import { markdownPreviewProvider } from "../markdownPreviewProvider";
+import { CommandConstants } from "./commandConstants";
+
+/**
+ * Register the commands for vscode-xml
+ *
+ * @param context the extension context
+ */
+export async function registerCommands(context: ExtensionContext, languageClient: LanguageClient) {
+
+  // XML Documentation commands
+  context.subscriptions.push(markdownPreviewProvider);
+  context.subscriptions.push(commands.registerCommand(CommandConstants.OPEN_DOCS_HOME, async () => {
+    const uri = 'README.md';
+    const title = 'XML Documentation';
+    const sectionId = '';
+    markdownPreviewProvider.show(context.asAbsolutePath(path.join('docs', uri)), title, sectionId, context);
+  }));
+  context.subscriptions.push(commands.registerCommand(CommandConstants.OPEN_DOCS, async (params: { page: string, section: string }) => {
+    const page = params.page.endsWith('.md') ? params.page.substr(0, params.page.length - 3) : params.page;
+    const uri = page + '.md';
+    const sectionId = params.section || '';
+    const title = 'XML ' + page;
+    markdownPreviewProvider.show(context.asAbsolutePath(path.join('docs', uri)), title, sectionId, context);
+  }));
+
+  // Code Lens actions
+  context.subscriptions.push(commands.registerCommand(CommandConstants.SHOW_REFERENCES, (uriString: string, position: Position) => {
+    const uri = Uri.parse(uriString);
+    workspace.openTextDocument(uri).then(document => {
+      // Consume references service from the XML Language Server
+      let param = languageClient.code2ProtocolConverter.asTextDocumentPositionParams(document, position);
+      languageClient.sendRequest(ReferencesRequest.type, param).then(locations => {
+        commands.executeCommand(CommandConstants.EDITOR_SHOW_REFERENCES, uri, languageClient.protocol2CodeConverter.asPosition(position), locations.map(languageClient.protocol2CodeConverter.asLocation));
+      })
+    })
+  }));
+
+  // Revalidate current file
+  context.subscriptions.push(commands.registerCommand(CommandConstants.VALIDATE_CURRENT_FILE, async (params) => {
```

Comment:
Create a new function registerValidateCommands and move the validate code to this function